### PR TITLE
Bump versions, address deprecations

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
The general idea of this PR is making console output cleaner. This PR:
  * Bump dependencies on Spine artifacts.
  * Addresses Gradle warnings by adding explicit dependencies between tasks.
  * Rolls back the Protobuf dependency to version `3.19.6` to overcome the missing artefact issue under macOS running on M1.
  * Adds ErrorProne checks to the build.

### CODE REVIEW

File: buildSrc/src/main/kotlin/deps-between-tasks.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     *
+     * This extension is kept local to `configureTaskDependencies` extension
+     * to prevent its direct usage from outside.
+     */
+    fun String.dependOn(taskName: String) {
+        val whoDepends = this
+        val dependOntoTask: Task? = tasks.findByName(taskName)
+        dependOntoTask?.apply {
+            tasks.findByName(whoDepends)?.dependsOn(this)
+        }
```
Comment (dmdashenkov) [inline buildSrc/src/main/kotlin/deps-between-tasks.kt:55]: Please use an `if` or a `let` instead. The meaning of `this` is not 100% clear from the first glance.

Comment (dmdashenkov) [reply buildSrc/src/main/kotlin/deps-between-tasks.kt:55]: This is how we get JavaScript :)

**Review by dmdashenkov — APPROVED** [review]
LGTM with a suggestion.

File: buildSrc/src/main/kotlin/deps-between-tasks.kt (hunk 1 above)
Comment (alexander-yevsyukov) [reply buildSrc/src/main/kotlin/deps-between-tasks.kt:55]: Goot catch. I'll update `config` and `template` too then.

**Pull request merged**